client/text.cpp

```diff
@@ -1400,128 +1401,256 @@ QString text_happiness_wonders(const struct city *pcity)
   return str.trimmed();
 }
 
+namespace /* anonymous */ {
+
 /**
-   Describing city factors that affect happiness.
+ * Describes the rules of city happiness for the given player.
  */
-const QString text_happiness_cities(const struct city *pcity)
+QString text_happiness_cities_rules(const player *pplayer, int base_content,
+                                    int basis, int step,
+                                    bool depends_on_empire_size)
 {
-  struct player *pplayer = city_owner(pcity);
-  int cities = city_list_size(pplayer->cities);
-  int content = get_player_bonus(pplayer, EFT_CITY_UNHAPPY_SIZE);
-  int basis = get_player_bonus(pplayer, EFT_EMPIRE_SIZE_BASE);
-  int step = get_player_bonus(pplayer, EFT_EMPIRE_SIZE_STEP);
-  QString str;
+  auto str = QString();
 
-  if (basis + step <= 0) {
-    /* Special case where penalty is disabled; see
-     * player_content_citizens(). */
-    str += QString(PL_("Cities: %1 total, but no penalty for empire size.",
-                       "Cities: %1 total, but no penalty for empire size.",
-                       cities))
-               .arg(QString::number(cities))
-           + qendl();
-    // TRANS: %d is number of citizens
-    str +=
-        QString(PL_("%1 content per city.", "%1 content per city.", content))
-            .arg(QString::number(content))
-        + qendl();
+  bool next_citizens_are_angry = false;
+  if (base_content == 0) {
+    str += _("All cities start with all citizens unhappy.");
+    next_citizens_are_angry = game.info.angrycitizen;
   } else {
-    // Can have up to and including 'basis' cities without penalty
-    int excess = MAX(cities - basis, 0);
-    int penalty;
-    int unhappy, angry;
-    int last, next;
-
-    if (excess > 0) {
-      if (step > 0) {
-        penalty = 1 + (excess - 1) / step;
-      } else {
-        penalty = 1;
+    str += QString(PL_("All cities start with %1 content citizen.",
+                       "All cities start with %1 content citizens.",
+                       base_content))
+               .arg(base_content);
+  }
+  if (depends_on_empire_size) {
+    if (basis > 0) {
+      str += QStringLiteral(" ");
+      if (next_citizens_are_angry) {
+        str += QString(PL_("Once you have more than %1 city, a citizen "
+                           "becomes angry.",
+                           "Once you have more than %1 cities, a citizen "
+                           "becomes angry.",
+                           basis))
+                   .arg(basis);
+      } else if (base_content == 1) {
+        // TRANS: Comes after "All cities start with 1 content citizen."
+        //        "it" is the citizen.
+        str +=
+            QString(
+                PL_("Once you have more than %1 city, it becomes unhappy.",
+                    "Once you have more than %1 cities, it becomes unhappy.",
```

Comment:
instead of using "it", maybe be more explicit and say "a citizen becomes unhappy"

Reply:
This is the case for "All cities start with 1 content citizen. Once you have more than 5 cities, it becomes unhappy." The general case is handled below and reads as you suggested